examples/serialize.rs

```diff
@@ -8,7 +8,7 @@ fn main() {
 
     let arr = vec![
         Bson::String("blah".to_string()),
-        Bson::DateTime(chrono::Utc::now()),
+        Bson::DateTime(chrono::Utc::now().into()),
```

Comment:
maybe not technically possible, but figured I'd ask in case there's some way to be clever, with this change is there any way around users having to add the `into()` calls? (like maybe you can call that for them automatically in a constructor or something to avoid the clunkiness?)

on the other hand this sounds vaguely familiar to a recent decision we made.... somewhere... to avoid hiding allocations from the user, which I think we would be doing if we did this. so maybe it's good to make them know a conversion is happening explicitly

Reply:
We could introduce a `Bson::from_chrono_0_4_datetime` constructor method which would avoid the need for an `.into()` call, but that's probably less ergonomic than using the `From`/`Into` trait.

An alternative to that would be to use feature flags to dictate which chrono version we're using, and then depending on that flag just have the `chrono::DateTime` directly in the enum variant. I only mentioned this briefly in the doc though I probably should have included a more detailed example.
```rust
pub enum Bson {
     // other cases ...

     #[cfg(feature = "chrono_0_4")]
     DateTime(chrono_0_4::DateTime),

     #[cfg(feature = "chrono_0_5")]
     DateTime(chrono_0_5::DateTime),

    // rest of cases...
}
```

The problem here is we'll require that a user select a feature, which could be kind of annoying. We could default to the current implementation in this PR if no feature is specified, though then we'd be presenting two slightly different APIs that may make documentation or examples hard. A better default would probably be to enable the "chrono_0_4" feature by default and just add a new "chrono" feature if/when chrono does 1.0. Once that happens the `Cargo.toml` invocation for the driver would potentially be kind of annoying:

```toml
mongodb = { default-features = false, features = ["tokio-runtime", "bson/chrono"], version = "2" }
```
For async-std users this won't be much of a difference though.

Reply:
I think the initial decision we made (solution 1 over 2) still seems better from a stability perspective, since as you mention in the doc driver 2.0 may outlive chrono 0.4. given that, I think the minor ergonomic tradeoff of calling `From`/`Into` seems fine, and I agree a separate BSON constructor to avoid the call probably doesn't provide a ton of value